pkg/eval/builtin_special.go

```diff
@@ -331,23 +336,30 @@ func use(fm *Frame, spec string, r diag.Ranger) (*Ns, error) {
 		path := filepath.Clean(dir + "/" + spec)
 		return useFromFile(fm, spec, path, r)
 	}
+
+	// Handle imports of pre-defined modules like `builtin` and `str`.
 	if ns, ok := fm.Evaler.modules[spec]; ok {
 		return ns, nil
 	}
 	if code, ok := fm.Evaler.BundledModules[spec]; ok {
 		return evalModule(fm, spec,
 			parse.Source{Name: "[bundled " + spec + "]", Code: code}, r)
 	}
+
+	// Handle imports relative to the Elvish config lib directories.
+	//
 	// TODO: For non-relative imports, use the spec (instead of the full path)
 	// as the module key instead to avoid searching every time.
 	for _, dir := range fm.Evaler.LibDirs {
 		ns, err := useFromFile(fm, spec, filepath.Join(dir, spec), r)
-		if _, isNoSuchModule := err.(noSuchModule); isNoSuchModule {
+		if _, isNoSuchModule := err.(NoSuchModule); isNoSuchModule {
 			continue
 		}
 		return ns, err
 	}
-	return nil, noSuchModule{spec}
+
+	// Sadly, we couldn't resolve the module spec.
```

Comment:
This comment doesn't say anything that's not clear from the code.

Some codebases use a pseudo-literate style where the comments form a complete description of the behavior of the code, but Elvish's codebase does not use that style and I don't intend to adopt that either.

Reply:
Fair enough. Removed. I added that comment to be explicit about the logic but agree it isn't needed for any competent contibutor.